vowpalwabbit/cats_tree.h

```diff
@@ -54,6 +54,7 @@ struct node_cost
 {
   uint32_t node_id;
   float cost;
+  node_cost(uint32_t node_id = 0, float cost = 0) : node_id(node_id), cost(cost) {}
```

Comment:
This is nit, so feel free to wontfix. I'd say updating lines 55 and 56 is preferable as default params in constructors is a but confusing and the default constructor compared to this constructor with default parameters is ambiguous.  (well default won't be generated in this case, and I think default default constructor is usually best)
```c++
uint32_t node_id = 0;
float cost = 0.f;
```

Reply:
I actually tried this initially, however it causes a compilation error with commands such as "_a = {node_id, _cost_star};"

Reply:
I see. In that case I think we should have two constructors. Default and two parameter rather than a since one with default parameters.

Reply:
Ok just recommitted.